pkg/controller/storagecluster/volumesnapshotterclasses.go

```diff
@@ -46,30 +52,38 @@ func newVolumeSnapshotClass(instance *ocsv1.StorageCluster, snapShotterType Snap
 	return retSC
 }
 
-func newSnapshotClasses(instance *ocsv1.StorageCluster) []*snapapi.VolumeSnapshotClass {
-	scs := []*snapapi.VolumeSnapshotClass{
-		newVolumeSnapshotClass(instance, cephfsSnapshotter),
-		newVolumeSnapshotClass(instance, rbdSnapshotter),
+func newCephFilesystemSnapshotClass(instance *ocsv1.StorageCluster) *SnapshotClassConfiguration {
+	return &SnapshotClassConfiguration{
+		snapshotClass:     newVolumeSnapshotClass(instance, cephfsSnapshotter),
+		reconcileStrategy: ReconcileStrategy(instance.Spec.ManagedResources.CephFilesystems.ReconcileStrategy),
+		disable:           instance.Spec.ManagedResources.CephFilesystems.DisableSnapshotClass,
 	}
-	return scs
 }
 
-func (r *ReconcileStorageCluster) createSnapshotClasses(vscs []*snapapi.VolumeSnapshotClass, instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
-	for index, vsc := range vscs {
-		reconcileStrategy := ReconcileStrategyIgnore
-		disableSnapshotClass := false
-		switch index {
-		case cephFileSystemIndex:
-			reconcileStrategy = ReconcileStrategy(instance.Spec.ManagedResources.CephFilesystems.ReconcileStrategy)
-			disableSnapshotClass = instance.Spec.ManagedResources.CephFilesystems.DisableSnapshotClass
-		case cephBlockPoolIndex:
-			reconcileStrategy = ReconcileStrategy(instance.Spec.ManagedResources.CephBlockPools.ReconcileStrategy)
-			disableSnapshotClass = instance.Spec.ManagedResources.CephBlockPools.DisableSnapshotClass
-		}
-		if reconcileStrategy == ReconcileStrategyIgnore || disableSnapshotClass {
+func newCephBlockPoolSnapshotClass(instance *ocsv1.StorageCluster) *SnapshotClassConfiguration {
+	return &SnapshotClassConfiguration{
+		snapshotClass:     newVolumeSnapshotClass(instance, rbdSnapshotter),
+		reconcileStrategy: ReconcileStrategy(instance.Spec.ManagedResources.CephFilesystems.ReconcileStrategy),
+		disable:           instance.Spec.ManagedResources.CephFilesystems.DisableSnapshotClass,
+	}
+}
+
+func newSnapshotClasses(instance *ocsv1.StorageCluster) []*SnapshotClassConfiguration {
```

Comment:
This and all other builder functions(i.e new<--->() ) should be renamed to reflect the fact that it now returns `SnapshotClassConfiguration` which consists of `SnapshotClass` and other properties.